### PR TITLE
Fill in docs for config on edges, and fix logging statements

### PR DESCRIPTION
**Pull request by drwill-ms**
#2419

### CODE REVIEW

File: iothub/service/src/Registry/RegistryManager.cs [hunk 1]
```diff
@@ -2110,6 +2216,7 @@ public virtual Task<Twin> ReplaceTwinAsync(string deviceId, string moduleId, str
         /// </summary>
         /// <param name="configuration">The Configuration object being registered.</param>
         /// <returns>The Configuration object.</returns>
+        /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/>
```
Comment (drwill-ms) [inline iothub/service/src/Registry/RegistryManager.cs:2219]: Adding a relevant web topic for managing configurations.

File: iothub/service/src/Registry/RegistryManager.cs [hunk 2]
```diff
@@ -2392,47 +2528,58 @@ public virtual Task RemoveConfigurationAsync(Configuration configuration, Cancel
...
             {
-                Logging.Exit(this, $"Removing configuration: {configuration?.Id}", nameof(RemoveConfigurationAsync));
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Removing configuration: {configuration?.Id}", nameof(RemoveConfigurationAsync));
             }
         }
 
         /// <summary>
-        /// Applies configuration content to a device.
+        /// Applies configuration content to an Edge device to create a deployment.
```
Comment (drwill-ms) [inline iothub/service/src/Registry/RegistryManager.cs:2543]: Key documentation updates starting here. Also in `<remarks>`.

**Review by brycewang-microsoft — APPROVED** [review]

**Review by Marusyk — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

File: iothub/service/src/Configurations/ConfigurationContent.cs [hunk 3]
```diff
@@ -17,6 +17,9 @@ public class ConfigurationContent
         /// <remarks>
         /// See <see href="https://docs.microsoft.com/azure/iot-edge/module-composition?view=iotedge-2020-11#create-a-deployment-manifest"/>
         /// and <see href="https://docs.microsoft.com/azure/iot-hub/iot-hub-automatic-device-management"/> for more details.
+        /// <para>
+        /// For Edge deployments, this should have a key of <c>"$edgeAgent"</c>.
+        /// </para>
```
[the remark this answers is not in the document]
Comment (drwill-ms) [reply iothub/service/src/Configurations/ConfigurationContent.cs:22]: I'll ask the service team contacts I'm working with on this.

**Pull request merged**